application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java

```diff
@@ -133,50 +141,60 @@ public void testTour() throws Exception
         homePage.nextStep();
         assertEquals("Title 3", homePage.getStepTitle());
         assertEquals("Step 3", homePage.getStepDescription());
-        assertFalse(homePage.hasNextStep());
+        assertTrue(homePage.hasNextStep());
         assertTrue(homePage.hasPreviousStep());
-        assertTrue(homePage.hasEndButton());
+        assertFalse(homePage.hasEndButton());
+
+        // Step 4
+        // Use a second page to test the Multipage feature
+        PageWithTour secondPage = PageWithTour.gotoPage("TourCode", "TourClass");
```

Comment:
A general issue is that if you visit a page that is not the target page for the current step , the tour redirects you to the page of the step.
To answer to your question: If you visit first `TourClass`, you'll be redirected to the homepage at the first step, but having the `Show tour` button.